src/Files.App/Views/LayoutModes/StandardLayoutMode.cs

```diff
@@ -194,27 +197,60 @@ protected virtual void RenameTextBox_LostFocus(object sender, RoutedEventArgs e)
 
 		protected void RenameTextBox_KeyDown(object sender, KeyRoutedEventArgs e)
 		{
-			if (e.Key == VirtualKey.Escape)
+			var textBox = (TextBox)sender;
+			switch (e.Key)
 			{
-				TextBox textBox = (TextBox)sender;
-				textBox.LostFocus -= RenameTextBox_LostFocus;
-				textBox.Text = OldItemName;
-				EndRename(textBox);
-				e.Handled = true;
-			}
-			else if (e.Key == VirtualKey.Enter)
-			{
-				TextBox textBox = (TextBox)sender;
-				textBox.LostFocus -= RenameTextBox_LostFocus;
-				CommitRename(textBox);
-				e.Handled = true;
+				case VirtualKey.Escape:
+					textBox.LostFocus -= RenameTextBox_LostFocus;
+					textBox.Text = OldItemName;
+					EndRename(textBox);
+					e.Handled = true;
+					break;
+				case VirtualKey.Enter:
+					textBox.LostFocus -= RenameTextBox_LostFocus;
+					CommitRename(textBox);
+					e.Handled = true;
+					break;
+				case VirtualKey.Up:
+					textBox.SelectionStart = 0;
+					e.Handled = true;
+					break;
+				case VirtualKey.Down:
+					textBox.SelectionStart = textBox.Text.Length;
+					e.Handled = true;
+					break;
+				case VirtualKey.Left:
+					e.Handled = textBox.SelectionStart == 0;
+					break;
+				case VirtualKey.Right:
+					e.Handled = (textBox.SelectionStart + textBox.SelectionLength) == textBox.Text.Length;
+					break;
+				case VirtualKey.Tab:
+					textBox.LostFocus -= RenameTextBox_LostFocus;
+					textBox.Text = OldItemName;
+					EndRename(textBox);
```

Comment:
Shouldn't this commit the name?

Reply:
Yes, it make sense. I'll need some time to implement it, since committing a name causes the file list to refresh